### PR TITLE
FSE: Update NUX slides

### LINKED ISSUE #41960: Update the NUX modal
Let's update the text and images for the Gutenboarding NUX modal! 🙌

The changes require work in the [FSE plugin directory](https://github.com/Automattic/wp-calypso/blob/ccf9cf2f21b9038274d1278e99129980fb5568e0/apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js), and are meant for Gutenboarding sites only. 

The "Private until you're ready to launch" slide is [commented](https://github.com/Automattic/wp-calypso/blob/ccf9cf2f21b9038274d1278e99129980fb5568e0/apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js#L102-L111) out at present for all sites, but we can turn it back on for Gutenboarding sites.

### The copy

> **Welcome to your website**
> Edit your homepage, add the pages you need, and change your site’s look and feel.
> 
> **Add or edit your content**
> Edit the placeholder content we’ve started you off with, or click the plus sign to add more content.
> 
> **Preview your site as you go**
> As you edit your site content, click “Preview” to see your site the way your visitors will.
> 
> **Private until you’re ready**
> Your site will remain private as you make changes until you’re ready to launch and share with the world.

image
image
image
image

Context and links to Figma files
p1588879262397200-slack-CPFFQTG69

### PR DESCRIPTION
**Pull request by ramonjd**
## Changes proposed in this Pull Request

We're updating the copy and image based on the description in #41960

After unsuccessful attempts to render `<NuxPage />` inside `<Guide />` conditionally^, I went with a filtered collection approach.

All suggestions for improvement wanted. :)

## Testing instructions

Easiest way I found is to check out this branch

Sandbox a test site

Comment out [the following lines](https://github.com/Automattic/wp-calypso/blob/master/apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js#L47):

```
	/* if ( ! isWpcomNuxEnabled || isSPTOpen ) {
		return null;
	} */
```

Run `cd apps/full-site-editing && yarn dev --sync`

Test Gutenboarding and non-Gutenboarding versions by setting `isGutenboarding` to `true/false`

#### isGutenboarding === true
isGutenbergNUX

#### isGutenboarding === false
notGutenbergNUX

Fixes #41960

^ The result of trying something like `{ isGutenboarding && <NuxPage { ...props } /> }`
nux-oops

### CODE REVIEW

**Comment by ramonjd** [comment]
> When I look at the site in spanish, they are not translated! But actually that's the same with the onboarding text before this PR too

Oh! Thanks for pointing that out. I'm going to double check. Probably a good argument for just replacing the NUX journey completely then with @joanrho's new designs/copy.

**Comment by simison** [comment]
Hmm, indeed, I don't find some of those older strings from translation tool at all. That's worrying!

**Comment by simison** [comment]
>Could it be they need textdomain?

Looks like that: https://github.com/Automattic/wp-calypso/search?q=full-site-editing&unscoped_q=full-site-editing

**Comment by ramonjd** [comment]
> Could it be they need textdomain? __( 'Example', 'full-site-editing' )?

Oh! 🤦 

I added the domain in the master branch code just to check, but didn't see any changes:

Screen Shot 2020-05-08 at 6 54 32 pm

Funny that the buttons don't have the text domain, but they are translated

**Comment by akirk** [comment]
Inside Calypso we don't need the text domain / ignore it. When in Gutenberg, translations are loaded via inlined JS in the HTML (using `wp_set_script_translations()`), there the text domain is important since the translations are loaded into the text domain. I'm unsure which context we're talking about here.

**Comment by simison** [comment]
This code would run in wp-admin context in Gutenberg, so sounds like we will need it.

Additionally, this (FSE plugin) code gets sent to .org repo and shipped as a plugin to Atomic sites. I'm not entirely sure where the translation happen but I'm assuming it happens at .org repo?

**Comment by ramonjd** [comment]
> I'm not entirely sure where the translation happen but I'm assuming it happens at .org repo?

As far as I can tell, the translations are all there, e.g., 

https://translate.wordpress.org/projects/wp-plugins/full-site-editing/dev/it/default/?filters%5Bterm%5D=Welcome+to+your+website&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=asc

https://translate.wordpress.org/projects/wp-plugins/full-site-editing/dev/fr/default/?filters%5Bterm%5D=Insert+text%2C+photos%2C+forms%2C+Yelp+reviews%2C+testimonials%2C+maps%2C+and+more.+Rearrange+the+blocks+on+your+page+until+they%E2%80%99re+just+right&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=asc

I'll add the domain arg to the translations.

**Comment by ramonjd** [comment]
I've also made the new copy the default as per the comments over at https://github.com/Automattic/wp-calypso/issues/41960#issuecomment-625548330

We were originally hiding the privacy slide so I've kept that for now for non-Gutenboarding sites until further instruction

**Comment by simison** [comment]
> We were originally hiding the privacy slide so I've kept that for now for non-Gutenboarding sites until further instruction

Yup, that's good. That's because users who have old published sites also can see the modal.

File: apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js [hunk 1]
```diff
@@ -50,69 +50,72 @@ function WpcomNux() {
...
-				heading={ __( "Private until you're ready to launch" ) }
-				description={ __(
-					"Your site remains private until you're ready to launch. Take your time and make as many changes as you need until it's ready to share with the world."
-				) }
-				imgSrc={ privateImage }
-				alignBottom
-			/>
-			*/ }
+			{ getWpcomNuxPages( isGutenboarding ).map( ( nuxPage ) => (
+				<NuxPage { ...nuxPage } />
```
Comment (sirreal) [inline apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js:64]: This doesn't complain about missing `key`?

Comment (simison) [reply apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js:64]: We test on wp-admin where these warnings don't seem to show up.

Comment (simison) [reply apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js]: Added heading as the key in https://github.com/Automattic/wp-calypso/pull/41966/commits/c0af4f691f73380ab3c336f3fdb462a459a55758

**Comment by simison** [comment]
Rebased to get `build-fse-plugin / Build FSE plugin` step pass.

File: apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js (hunk 1 above)
Comment (ramonjd) [reply apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nux/src/wpcom-nux.js]: > This doesn't complain about missing key?

Doh, forgot the key. Thanks for adding.

**Pull request merged**